py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -47,26 +47,16 @@ def select(
     if dynamic_shape:
         # Check whether slice target dim is dynamic shape dim
         assert input.shape[dim] != -1, "Can't select on negative shape dimension!"
-    index = index
 
     if index >= input.shape[dim]:
         raise RuntimeError(
             f"cannot have index greater than the dimension length! {input.shape[dim]}"
         )
-    output_shape = list(input.shape)
-    output_shape[dim] = 1
-    if dynamic_shape > 0:
-        output_shape = get_shape_with_dynamic_shape(
-            ctx, target, source_ir, name, output_shape, input
-        )
+
```

Comment:
If above asserts() are removed and full dynamic shape was used(e.g.(-1,-1,-1)), test worked.
I'm wondering if select on dynamic dim can be supported.

Reply:
I have tested the test you mentioned and confirmed that using -1, -1, -1 also passes the test cases successfully.

However, I didn't remove the asserts() because the test cases fail when the `index` is out of range, meaning the `index` is larger than the corresponding dimension in the dynamic input shape. 

```
            (
                "success case",
                (1, 1, 1),
                (2, 2, 2),
                (3, 3, 3),
                torch.float,
                0,
                1,
            ),
            (
                "fail case",
                (1, 1, 1),
                (2, 2, 2),
                (3, 3, 3),
                torch.float,
                0,
                3,
            ),
```

It seems that modifying the `index` like dimension (`dim = get_positive_dim(cast(int, dim), ranks)`) when the `index` is greater than the `size` would solve the issue. Do you have an example that handles it this way?

Reply:
I think we cannot check invalid index for dynamic dim. Error will happen in runtime.
Maybe we can check index for only static dim.
if DYNAMIC_DIM != input.shape[dim] and index >= input.shape[dim]:

Reply:
I also tested by removing the `assert()` and `raise RuntimeError()` statements and tested with (-1, -1, -1). 

The test case mentioned as a `success case` above passes, but the test case mentioned as a `fail case` fails. The reason for this is that in the `fail case`, the size of the 0-th dimension is 3, but the `index` selects positions larger than that (`index`=3, 4th position). The current converter does not handle this case where the `index` is larger than the size of the dimension specified by `dim`, as in the 'fail case'. (Note: It might be possible to handle this using the slice layer, lt (less than) and div functions, but currently, it is handled with an assert statement.) 

Therefore, I left the `assert()` and `raise RuntimeError()` statement without removing it.

Reply:
Oh! I misunderstood. If the `index` is larger than the size of the dimension specified by `dim`, it won't work in PyTorch either, so we don't need to handle this case. Therefore, we don't need to consider test cases like the 'fail case' mentioned above. This means that a dynamic shape is supported for all dimensions. 

<Example of correct usage of select.int in normal PyTorch>
![image](https://github.com/user-attachments/assets/d29f55f8-c4e5-46ce-b6a6-65ff5bc9ba85)
<Example of IndexError occurring in PyTorch when using select.int>
![image](https://github.com/user-attachments/assets/3b5a1356-8580-474f-a151-02151ead9bc8)

Reply:
@keehyuna Thanks for the suggestion!

Since PyTorch already raises an error when the index exceeds the input size, there’s no need for us to check this here. I removed those checks. Additionally, following your suggestion, I’ve added test cases to fully support dynamic shapes.